runtime/mount/python_lib/lplots/h5/h5ad/process_message.py

```diff
@@ -490,13 +503,31 @@ async def process_h5ad_request(
         image_uri = await fetch_and_process_image(
             msg["node_id"], msg["s3_presigned_url"]
         )
+
+        image_id = msg["id"]
+
+        images = adata.uns.get("latch_images", {})
+        image_data = images.get(image_id, {})
+
+        image_data["b64_image"] = image_uri
+        image_data["node_id"] = msg["node_id"]
+        image_data["id"] = image_id
+        image_data["name"] = msg["name"]
+        image_data["transformations"] = msg["transformations"]
+        images[image_id] = image_data
+        adata.uns["latch_images"] = images
+
         return {
             "type": "h5",
             "op": op,
             "data_type": "h5ad",
             "key": widget_session_key,
             "value": {
-                "data": {"image": image_uri, "fetched_for_node_id": msg.get("node_id")}
+                "data": {
+                    "image_data": image_data,
+                    "image_id": image_id,
+                    "fetched_for_node_id": msg.get("node_id"),
+                }
```

Comment:
this should probably be a fetched_for_image_id now right and we can drop passing back the image_id?